sdk/core/core-rest-pipeline/test/node/userAgentPlatform.spec.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { describe, it, assert, vi, afterEach } from "vitest";
+import { setPlatformSpecificData } from "../../src/util/userAgentPlatform.js";
+import * as process from "process";
+
+describe("userAgentPlatform", () => {
+  vi.mock("process", async () => {
+    const actual = await vi.importActual("process");
+    return {
+      ...actual,
+      versions: {},
+    };
+  });
+
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should handle an empty process.versions", () => {
```

Comment:
Can we add a test where `process.versions` is undefined instead of empty?

Reply:
Changed to undefined